server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -595,6 +596,7 @@ protected Node(final Environment initialEnvironment, Collection<PluginInfo> clas
 
             runnableTaskListener = new AtomicReference<>();
             final ThreadPool threadPool = new ThreadPool(settings, runnableTaskListener, executorBuilders.toArray(new ExecutorBuilder[0]));
+            threadPool.getThreadContext().registerThreadContextStatePropagator(new WorkloadGroupThreadContextStatePropagator());
```

Comment:
I think a better way would be to do this inside `WorkloadGroupThreadContextStatePropagator`?

```
WorkloadGroupThreadContextStatePropagator(ThreadContext threadContext) {
     threadContext. registerThreadContextStatePropagator(this);
}

Reply:
Can we also add a basic test to verify this functionality?

Reply:
I will add a UT which will confirm the behavior

Reply:
> I think a better way would be to do this inside WorkloadGroupThreadContextStatePropagator?

Why do we think that is a better way, In the end we will still have to init the `WorkloadGroupThreadContextStatePropagator` somewhere